test/app.test.js

```diff
@@ -288,6 +288,209 @@ describe("app", () => {
+            children: [
+              h("li", {key: "red"}, ["red"]),
+              h("li", {key: "orange"}, ["orange"]),
+              h("li", {}, ["unkeyed"]),
+              h("li", {key: "yellow"}, ["yellow"]),
+              h("li", {}, ["unkeyed"]),
+              h("li", {key: "green"}, ["green"]),
+              h("li", {key: "blue"}, ["blue"]),
+            ]
+          },
+          {
+            name: "Add two disjoint unkeyed nodes in the middle",
+            children: [
+              h("li", {key: "red"}, ["red"]),
+              h("li", {}, ["unkeyed"]),
+              h("li", {key: "orange"}, ["orange"]),
+              h("li", {key: "yellow"}, ["yellow"]),
+              h("li", {}, ["unkeyed"]),
+              h("li", {key: "green"}, ["green"]),
+              h("li", {}, ["unkeyed"]),
+              h("li", {key: "blue"}, ["blue"]),
+            ]
+          },
+          {
+            name: "Remove two disjoint nodes from the middle",
+            children: [
+              h("li", {key: "red"}, ["red"]),
+              h("li", {}, ["unkeyed"]),
+              h("li", {key: "yellow"}, ["yellow"]),
+              h("li", {key: "blue"}, ["blue"]),
+            ]
+          },
+        ]
+
+        tests.forEach(test => {
+          it(test.name, function (done) {
+            app({
+              model: true,
+              actions: { toggle: model => !model },
+              subscriptions: [ (_, actions) => setTimeout(_ => actions.toggle(), 0) ],
+              view: model => {
+                function setColors(el) {
+                  for (var i = 0; i < el.childNodes.length; i++) {
+                    if (el.childNodes[i].textContent === "unkeyed") continue
+                    el.childNodes[i].style.color = el.childNodes[i].textContent
+                  }
+                }
+                function checkColors(el) {
+                  for (var i = 0; i < el.childNodes.length; i++) {
+                    if (el.childNodes[i].textContent === "unkeyed") continue
+                    if (el.childNodes[i].style.color === el.childNodes[i].textContent) continue
+                    fail("a keyed child was destroyed")
+                  }
+                  done()
+                }
+                if (model) {
+                  return h("ul", {onCreate: setColors}, [
+                    h("li", {key: "red"}, ["red"]),
+                    h("li", {key: "orange"}, ["orange"]),
+                    h("li", {key: "yellow"}, ["yellow"]),
+                    h("li", {}, ["unkeyed"]),
+                    h("li", {key: "green"}, ["green"]),
+                    h("li", {key: "blue"}, ["blue"]),
+                  ])
+                } else {
+                  return h("ul", {onUpdate: checkColors}, test.children)
+                }
+              }
+            })
+          })
+        })
+      })
+
+      it("fires onremove for ONLY the not-reused keyed nodes", done => {
```

Comment:
Ok, I'll remove them. But can we wait until it's definitely time to merge? Because they're pretty useful for knowing I don't break stuff when I fix merge-conflicts.